### PR TITLE
Add hamburger menu and mobile style changes

### PR DESCRIPTION
**Pull request by cvrrie**
### Tickets:

- HCK-47, HCK-49

### List of changes:

- Add hamburger styles
- Fix mobile padding to be unscrollable and consistent
- Make navbar sticky
- Style media queries for navbar and footer
- Style media queries for landing page
- Change email input font and styling
- Make footer icons bigger

### Type of change:

- [x] New feature (non-breaking change which adds functionality)

### How did you do this?
Added external component for hamburger menu

### PR Checklist:

- [x] Merged `develop` branch (before testing)
- [x] Linted my code locally
- [x] Tested all links in project relevant browsers
- [x] Tested all links on different screen sizes
- [x] Referenced all useful info (issues, tasks, etc)

### CODE REVIEW

File: src/components/nav.js [hunk 1]
```diff
@@ -1,21 +1,91 @@
 import React from "react"
-
+import { slide as Menu } from "react-burger-menu"
+import { Link } from "gatsby"
 import Logo from "../images/logo.svg"
 import * as Styles from "../styles/components/nav"
+import { Z_BUF_ERROR } from "zlib"
```
Comment (loreina) [inline src/components/nav.js]: what's this for?

File: src/styles/mailchimp.scss [hunk 2]
```diff
@@ -7,35 +7,29 @@ $mailchimp-color-white: #fff;
...
   font-weight: 400;
-  line-height: 27px;
+  line-height: 32px;
   color: $mailchimp-color-grey;
   outline: 0;
   border-width: 2px;
-  width: 400px;
-  padding-top: 7px; /* aligns input text in element */
+  width: 390px;
+  /*padding-top: 7px;  aligns input text in element */
```
Comment (loreina) [inline src/styles/mailchimp.scss]: Single-line comments should just use `//`

```suggestion
  // padding-top: 7px;  aligns input text in element
```

File: src/components/nav.js [hunk 3]
```diff
@@ -1,21 +1,34 @@
...
+import "../styles/hamburger.css"
```
Comment (loreina) [inline src/components/nav.js]: It looks like react-burger-menu supports JS styling, we should use that instead

File: src/styles/components/index.js [hunk 4]
```diff
@@ -6,14 +6,17 @@ const landingColorBorder = styleVars.hackBlack5
 const landingColorDarkBg = styleVars.hackBlack5
 
 export const Hero = styled.section`
-  min-height: 100vh;
+  height: 110%;
```
Comment (loreina) [inline src/styles/components/index.js]: why 110%?

File: src/components/hero.js [hunk 5]
```diff
@@ -44,7 +44,7 @@ const Hero = () => (
                   value=""
                   name="EMAIL"
                   class="required email"
-                  placeholder="your email here"
+                  placeholder="Your email here"
```
Comment (loreina) [inline src/components/hero.js]: Check the Figma design, it's lowercase

```suggestion
                  placeholder="your email here"
```

File: src/styles/mailchimp.scss [hunk 6]
```diff
@@ -7,35 +7,29 @@ $mailchimp-color-white: #fff;
...
-  text-align: center;
+  text-align: left;
   display: inline-block;
   text-transform: none;
   border-radius: 30px;
   border: 2px solid $mailchimp-color-main;
   transition: all 0.15s ease-in-out;
-
+  font-family: "Brown";
+  padding-left: 10px;
```
Comment (loreina) [inline src/styles/mailchimp.scss]: `padding-left` shifts the submit button on smaller screen sizes 

image

File: src/styles/hamburger.css [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+    top: 30px;
+    right: 30px;  
+}
+@media screen and (min-width: 768px) {
+    .bm-burger-button, .bm-menu-wrap {
+        display: none;
+    }
+}
+.bm-burger-bars {
+    background: #a6a6a6;
```
Comment (loreina) [inline src/styles/hamburger.css]: Add css color variables so we know which colours these are but should move to JS anyways

File: src/components/nav.js [hunk 8]
```diff
@@ -1,21 +1,34 @@
...
 
-const Nav = ({ scrollToAbout = () => {}, scrollToSponsor = () => {} }) => (
-  <Styles.NavBar>
-    <Styles.NavbarBrand>
-      <Styles.NavbarBrandImg src={Logo} />
-    </Styles.NavbarBrand>
-    <Styles.NavItems>
+const Nav = ({ scrollToAbout = () => {}, scrollToSponsor = () => {} }) => {
```
Comment (loreina) [inline src/components/nav.js]: Scroll doesn't work anymore?

File: src/styles/components/index.js [hunk 9]
```diff
@@ -132,7 +137,7 @@ export const HeroBackground = styled.img`
 
   @media only screen and (max-width: ${styleVars.smUp}) {
     position: relative;
-    max-width: 100%;
+    max-width: 80%;
```
Comment (loreina) [inline src/styles/components/index.js]: Just gonna trust you on these media query numbers

File: src/styles/hamburger.css [hunk 10]
```diff
@@ -0,0 +1,66 @@
+.bm-burger-button {
```
Comment (loreina) [inline src/styles/hamburger.css]: Mobile menu isn't centered

image

File: src/styles/mailchimp.scss (hunk 6 above)
Comment (cvrrie) [reply src/styles/mailchimp.scss]: I think I fixed it by doing box-sizing: border-box; and then playing around with the styles, hopefully it's good as a temporary fix

File: src/components/nav.js (hunk 3 above)
Comment (cvrrie) [reply src/components/nav.js]: I moved them all into the nav.js for styles

**Comment by loreina** [comment]
- HCK-47
- HCK-29
- HCK-52
- HCK-49
- HCK-86

**Pull request merged**